aten/src/ATen/native/cpu/ScatterGatherKernel.cpp

```diff
@@ -128,6 +128,54 @@ struct _cpu_scatter_gather_dim_loop {
   }
 };
 
+// dim loop for scatter when called on an index tensor
+// that has at least one dimension with index.size(d) > src.size(d)
+struct _cpu_scatter_large_index_dim_loop {
+  template <typename scalar_t, typename func_t>
+  void operator()(
+    scalar_t* self_data, int64_t self_dim_stride,
+    int64_t* index_data, int64_t index_dim_stride,
+    int64_t* index_starting_ptr, IntArrayRef index_shape,
+    IntArrayRef index_strides,
+    scalar_t* src_starting_ptr, IntArrayRef src_shape,
+    IntArrayRef src_strides,
+    int64_t dim, int64_t index_dim_size,
+    int64_t index_upper_bound,
+    func_t& f
+  ) {
+
+    for (const auto i : c10::irange(index_dim_size)) {
+      int64_t idx_dim = index_data[i * index_dim_stride];
+      // we are not putting idx_dim in the error message because it disables
+      // loop optimization in clang-7
+      TORCH_CHECK(idx_dim >= 0 && idx_dim < index_upper_bound,
+        "index ", index_data[i * index_dim_stride],
+        " is out of bounds for dimension ", dim,
+        " with size ", index_upper_bound
+      );
+
+      // calculate the equivalent indices in the source tensor based on
```

Comment:
As I understand it, from that comment, if we restrided the tensors in that way the entries of `index` that go beyond those in `src` would be ignored rather than looping back, which I think was the desired behaviour? Unless I'm missing something.

Reply:
Right, only indices in the "intersection" are considered.

Reply:
Hey @redmagnos, this is a good point! While the intersection is one possible interpretation that would be simpler to implement (which I believe is why the issue was labelled as a good first issue), you are right that the (much more technically complex) wraparound behavior that you have implemented seems to be what #63265 was suggesting judging by the code block under alternatives. 

Since the wraparound behavior doesn't seem to be working for noncontiguous tensors at the moment (per your comment below) perhaps we could hold off on fixing this for a bit while we determine which interpretation of the enhancement would best serve our users.

If you had a use-case for this wraparound behavior that motivated you to take up this issue, it would be great if you could share as well :)

Reply:
I don't, actually; I assumed the original poster themself would have a reason, but the reason I picked this was more "it was marked as a good first issue and no one else seemed to be actively working on it".

With respect to noncontiguous tensors, I implemented something that seems to work for them and had been testing them earlier today, so since the work is already done I'll push it and you guys can take a look. And in case the "intersection" solution is the preferred one after all, it should be reasonably easy to pivot in that direction (since it is in fact just a matter of restriding with no change in the logic elsewhere at all).